examples/secrets/values-secret.v2.yaml

```diff
@@ -0,0 +1,81 @@
+# NEVER COMMIT THESE VALUES TO GIT (unless your file only uses generated
+# passwords or only points to files)
+
+# Needed to specify the new format (missing version means old version: 1.0 by default)
+version: 2.0
+
+# These are the vault policies to be created in the vault
+# these are used when we let the vault generate the passwords
+# by setting the 'onMissingValue' attribute to 'generate'
+# See https://developer.hashicorp.com/vault/docs/concepts/password-policies
+vaultPolicies:
+  basicPolicy: |
+    length=10
+    rule "charset" { charset = "abcdefghijklmnopqrstuvwxyz" min-chars = 1 }
+    rule "charset" { charset = "ABCDEFGHIJKLMNOPQRSTUVWXYZ" min-chars = 1 }
+    rule "charset" { charset = "0123456789" min-chars = 1 }
+
+  advancedPolicy: |
+    length=20
+    rule "charset" { charset = "abcdefghijklmnopqrstuvwxyz" min-chars = 1 }
+    rule "charset" { charset = "ABCDEFGHIJKLMNOPQRSTUVWXYZ" min-chars = 1 }
+    rule "charset" { charset = "0123456789" min-chars = 1 }
+    rule "charset" { charset = "!@#$%^&*" min-chars = 1 }
+
+# This is the mandatory top-level secrets entry
+secrets:
+  # This will create the following keys + attributes:
+  # - secret/region-one/config-demo:
+  #     secret: ...<generated basicPolicy secret>...
+  #     secretprompt: ...<as input by the user>...
+  #     secretprompt2: ...<as input by the user. If just enter is pressed it will be 'defaultvalue'>...
+  #     secretfile: ...<content of the file as input by user. If just enter is pressed the file will be /tmp/ca.crt...0
+  #     ca_crt: ...<content of /tmp/ca.crt>...
+  #     ca_crt_b64: ...<content of /tmp/ca.crt base64-encoded before uploading to vault>...
+  # - secret/snowflake.blueprints.rhecoeng.com:
+  #     secret: ...<generated basicPolicy secret>...
+  #     secretprompt: ...<as input by the user>...
+  #     secretprompt2: ...<as input by the user. If just enter is pressed it will be 'defaultvalue'>...
+  #     secretfile: ...<content of the file as input by user. If just enter is pressed the file will be /tmp/ca.crt...0
+  #     ca_crt: ...<content of /tmp/ca.crt>...
+  #     ca_crt_b64: ...<content of /tmp/ca.crt base64-encoded before uploading to vault>...
+  - name: config-demo
+    vaultMount: secret
+    vaultPrefixes:
+    - region-one
+    - snowflake.blueprints.rhecoeng.com
+    fields:
+    - name: secret
+      onMissingValue: generate # One of: error,generate,prompt
+      vaultPolicy: basicPolicy
+    - name: secretprompt
+      value: null
+      onMissingValue: prompt
+      description: "Insert secretprompt NOW"
```

Comment:
Since people will cargo-cult the examples here, how about:
   "Please specify the (password|API key|...) used by ABC application to access the XYZ service"

Reply:
Ack